# Tarea 05: Cristian Tobar Morales
### Objetivo: Utilización de Git y librerias.

In [2]:
import pandas as pd # type: ignore
# Cargar desde un archivo CSV
df = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/iris-data.csv')
print(df.shape) # Mostrar las dimensiones
# 2. Mostrar las primeras (5) filas
df.head()


(150, 5)


sepal length  sepal width  petal length  petal width        class
0           5.1          3.5           1.4          0.2  Iris-setosa
1           4.9          3.0           1.4          0.2  Iris-setosa
2           4.7          3.2           1.3          0.2  Iris-setosa
3           4.6          3.1           1.5          0.2  Iris-setosa
4           5.0          3.6           1.4          0.2  Iris-setosa

##Qué columnas tiene el dataset y que representan.

Columna 1: Sepal length  
Columna 2: sepal width  
Columna 3: petal length  
Columna 4: petal_width  
Columna 5: species o clase  

Las columnas representan las longitudes o dimensiones de la especie, además se presenta la especie respectiva.


In [7]:
# La existencia (o no) de valores nulos.
df.info()
# No contiene existencia de valores nulos.

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal length  150 non-null    float64
 1   sepal width   150 non-null    float64
 2   petal length  150 non-null    float64
 3   petal width   150 non-null    float64
 4   class         150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


In [8]:
#Estadísticas de cada columna numérica (utilice df.describe()).
df.describe()

sepal length  sepal width  petal length  petal width
count    150.000000   150.000000    150.000000   150.000000
mean       5.843333     3.054000      3.758667     1.198667
std        0.828066     0.433594      1.764420     0.763161
min        4.300000     2.000000      1.000000     0.100000
25%        5.100000     2.800000      1.600000     0.300000
50%        5.800000     3.000000      4.350000     1.300000
75%        6.400000     3.300000      5.100000     1.800000
max        7.900000     4.400000      6.900000     2.500000

# Estadísticas de cada columna numérica
### Cantidad: 150 datos.
### Sepal length:
    - Media: 5.843  
    - Desviación Estandar: 0.828
    - Mínimo: 4.3
    - Máximo: 7.9
    - Percentil 25%: 5.1
    - Mediana: 5.8
    - Percentil 75%: 6.4

### Sepal width:
    - Media: 3.0540  
    - Desviación Estandar: 0.433
    - Mínimo: 2.0
    - Máximo: 4.4
    - Percentil 25%: 2.8
    - Mediana: 3.0
    - Percentil 75%: 3.3

### petal length:
    - Media: 3.758  
    - Desviación Estandar: 1.764
    - Mínimo: 1.0
    - Máximo: 6.9
    - Percentil 25%: 1.6
    - Mediana: 4.35
    - Percentil 75%: 5.1

### petal width:
    - Media: 1.198 
    - Desviación Estandar: 0.763
    - Mínimo: 0.10
    - Máximo: 2.5
    - Percentil 25%: 0.3
    - Mediana: 1.3
    - Percentil 75%: 1.8



In [8]:
#Cuántas filas existen para cada clase de flor.
df.groupby("class")["class"].count()


class
Iris-setosa        50
Iris-versicolor    50
Iris-virginica     50
Name: class, dtype: int64

#Cuántas filas existen para cada clase de flor.  
Existen 50 filas para cada especie de flor.